gridfs/grid_file.py

```diff
@@ -572,12 +575,14 @@ def read(self, size=-1):
         data.seek(0)
         return data.read(size)
 
-    def readline(self, size=-1):
+    def readline(self, size: Optional[int] = -1) -> bytes:
         """Read one line or up to `size` bytes from the file.
 
         :Parameters:
          - `size` (optional): the maximum number of bytes to read
         """
+        if size is None:
+            size = -1
```

Comment:
Why this change? The api does not support size=None and I'd like to avoid adding misc features in this PR.

Reply:
This is another case of `Optional[int]`.  In this case the superclass also uses it, we'd have to add an `# ignore: override`:

```
gridfs/grid_file.py:575: error: Argument 1 of "readline" is incompatible with supertype "IOBase"; supertype defines the argument type as "Optional[int]"
gridfs/grid_file.py:575: note: This violates the Liskov substitution principle
gridfs/grid_file.py:575: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

Reply:
Agreed, but because `io.IOBase` uses `Optional[int]`, we have to add an ignore override comment.  I'll do that.

Reply:
Oh I see. Yeah override is what typeshed does: https://github.com/python/typeshed/blob/df0fe1645681638359a4ab706732e79c62db6a71/stdlib/io.pyi#L128